### PR TITLE
Output alert applayer v13.4

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
https://redmine.openinfosecfoundation.org/issues/3827
https://redmine.openinfosecfoundation.org/issues/5977
https://redmine.openinfosecfoundation.org/issues/6500
https://redmine.openinfosecfoundation.org/issues/6501
https://redmine.openinfosecfoundation.org/issues/5053

Describe changes:
- Fix setup-app-layer script so that it adds app-layer metadata to alerts
- add krb5 metadata to alerts
- add ftp metadata to alerts
- add tftp metadata to alerts
- output/dns: do not add empty app-layer metadata
- app-layer: do not require probing parser as fixed patterns can be enough
- app-layer plugins

#9807 with S-V fixed for files

```
SV_BRANCH=pr/1465
```
https://github.com/OISF/suricata-verify/pull/1465

### CODE REVIEW

File: src/detect-engine-helper.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ */
+
+#include "suricata-common.h"
+#include "detect-engine.h"
+#include "detect-engine-helper.h"
+#include "detect-engine-mpm.h"
+#include "detect-engine-prefilter.h"
+#include "detect-parse.h"
+
+int DetectHelperBufferRegister(const char *name, AppProto alproto, bool toclient, bool toserver)
```
Comment (jasonish) [inline src/detect-engine-helper.c:32]: All this helper stuff seems generally useful across many keywords... Should it not be integrated directly into `detect-engine-register`?

Comment (catenacyber) [reply src/detect-engine-helper.c:32]: Good idea, will do

**Comment by catenacyber** [comment]
This also solves https://redmine.openinfosecfoundation.org/issues/4683

File: src/detect-engine-helper.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-engine-helper.c:32]: Problem is detect.h depends on detect-engine-register.h for DETECT_TBLSIZE_STATIC
So we cannot have these defined in detect-engine-register.h as they depend on definition in detect.h

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/9871

**Pull request closed without merging**